panic when putting non-record to .

$ zq -t "put .=key" input.tzng 
put .=x: cannot put a non-record to .
panic: put: unexpected step 0
goroutine 36 [running]:
runtime/debug.Stack(0xc0003a9ae8, 0xc0003a9ab8, 0x15a2869)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/brimsec/zq/zqe.RecoverError(0x17f3a60, 0xc000386180, 0xc000386180, 0x17f2520)
        /Users/henridf/work/brim/zq/zqe/error.go:163 +0x26
github.com/brimsec/zq/driver.(*mux).safeGet.func1(0xc0003a9f70)
        /Users/henridf/work/brim/zq/driver/mux.go:49 +0x48
panic(0x17f3a60, 0xc000386180)
        /usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/brimsec/zq/proc/put.step.build(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc0003ac101, 0xd, 0xf, 0xc00038e0a8, ...)
        /Users/henridf/work/brim/zq/proc/put/put.go:126 +0x33f
github.com/brimsec/zq/proc/put.(*Proc).put(0xc00038e090, 0xc0003940f0, 0xc0003940f0, 0x0, 0x0)
        /Users/henridf/work/brim/zq/proc/put/put.go:355 +0x185
github.com/brimsec/zq/proc/put.(*Proc).Pull(0xc00038e090, 0x0, 0x0, 0x0, 0x0)
        /Users/henridf/work/brim/zq/proc/put/put.go:369 +0x15f
github.com/brimsec/zq/driver.(*mux).safeGet(0xc0003840e0, 0x0, 0x0, 0x0, 0x0)
        /Users/henridf/work/brim/zq/driver/mux.go:51 +0x6c
github.com/brimsec/zq/driver.(*mux).run(0xc0003840e0)
        /Users/henridf/work/brim/zq/driver/mux.go:64 +0x32
created by github.com/brimsec/zq/driver.(*muxOutput).Pull.func1
        /Users/henridf/work/brim/zq/driver/mux.go:98 +0x5d

--- a/proc/put/put.go
+++ b/proc/put/put.go
@@ -348,8 +348,7 @@ func (p *Proc) put(in *zng.Record) (*zng.Record, error) {
 	}
 	rule, err := p.lookupRule(in.Type, vals)
 	if err != nil {
-		p.maybeWarn(err)
-		return in, nil
+		return nil, err
 	}
 
 	bytes, err := rule.step.build(in.Raw, &p.builder, vals)

--- a/proc/put/ztests/put-to-root-error-non-record.yaml
+++ b/proc/put/ztests/put-to-root-error-non-record.yaml
@@ -3,4 +3,5 @@ zql: 'put .=a'
 input: |
   #0:record[a:int32]
   0:[1;]
+  
 errorRE: cannot put a non-record to \.

--- a/proc/put/ztests/put-to-root-error-non-record.yaml
+++ b/proc/put/ztests/put-to-root-error-non-record.yaml
@@ -1,0 +1,6 @@
+zql: 'put .=a'
+
+input: |
+  #0:record[a:int32]
+  0:[1;]
+errorRE: cannot put a non-record to \.

